api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/BillServiceClient.java

```diff
@@ -56,6 +66,102 @@ public Flux<BillResponseDTO> getAllBilling() {
                 .bodyToFlux(BillResponseDTO.class);
     }
 
+    //to be changed
+//    public Flux<BillResponseDTO> getAllBillsByPage(Optional<Integer> page, Optional<Integer> size) {
+//        return webClientBuilder.build().get()
+//                .uri(billServiceUrl + "/bills-pagination?page="+page.orElse(0)+"&size="+size.orElse(5))
+//                .retrieve()
+//                .bodyToFlux(BillResponseDTO.class);
+//    }
+
+    public Flux<BillResponseDTO> getAllBillsByPage(Optional<Integer> page, Optional<Integer> size, String billId, String customerId,
+                                                    String ownerFirstName, String ownerLastName, String visitType,
+                                                    String vetId, String vetFirstName, String vetLastName) {
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(billServiceUrl + "/bills-pagination");
+
+        builder.queryParam("page", page);
+        builder.queryParam("size",size);
+
+        // Add query parameters conditionally if they are not null or empty
+        if (billId != null && !billId.isEmpty()) {
+            builder.queryParam("billId", billId);
+        }
```

Comment:
Could have been done using switch or if-else statements but this works fine as well!

Reply:
I didn't think about that, it could definitely work. My usual preference are if statements so I'll stick with that

Reply:
> Could have been done using switch or if-else statements but this works fine as well!

In this case an if-else is better since you might have more than one query param that are not provided.